vllm/executor/ray_utils.py

```diff
@@ -23,6 +26,21 @@ def __init__(self, *args, **kwargs) -> None:
             # that thread.
             self.compiled_dag_cuda_device_set = False
 
+            def dec_hook(type: Type, obj: Any) -> Any:
+                if type is array:
+                    deserialized = array('I')
+                    deserialized.frombytes(obj)
+                    return deserialized
+
+            def enc_hook(obj: Any) -> Any:
+                if isinstance(obj, array):
+                    # convert the complex to a tuple of real, imag
+                    return obj.tobytes()
```

Comment:
Question: why does array.tobytes convert a complex to a tuple of real, imag?

Reply:
just copied from example code lol